### ISSUE TITLE
Problem while installing in my Ubuntu 18.04 Server 

### ISSUE THREAD

**Issue by MiguelNdeCarvalho** [ONLYOFFICE/DocumentServer#478]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Latest Version

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
```
miguel@mike-machine:~$ sudo apt-get install onlyoffice-documentserver
Reading package lists... Done
Building dependency tree       
Reading state information... Done
onlyoffice-documentserver is already the newest version (5.2.7-2).
The following package was automatically installed and is no longer required:
  daemon
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] Y
Setting up onlyoffice-documentserver (5.2.7-2) ...
Generating AllFonts.js, please wait...Done
dpkg: error processing package onlyoffice-documentserver (--configure):
 installed onlyoffice-documentserver package post-installation script subprocess returned error exit status 1
Errors were encountered while processing:
 onlyoffice-documentserver
E: Sub-process /usr/bin/dpkg returned an error code (1)
miguel@mike-machine:~$ 
```

**What is the expected behavior?**
To install just fine xD

**Did this work in previous versions of DocumentServer?**
Didnt try
**Operating System:**Ubuntu Serer 18.04 LTS

I tried to follow this [guide](https://helpcenter.onlyoffice.com/server/linux/document/linux-installation.aspx). Thanks

**Comment by ShockwaveNN** [comment]
Maybe duplicate of https://github.com/ONLYOFFICE/DocumentServer/issues/271
Try to install `npm` and try again

**Comment by MiguelNdeCarvalho** [comment]
Hey mate, thanks for answering me. I have `npm` installed. This is the output when I use the command 

`miguel@mike-machine:~$ npm
Usage: npm <command>
where <command> is one of:
    access, adduser, audit, bin, bugs, c, cache, ci, cit,
    completion, config, create, ddp, dedupe, deprecate,
    dist-tag, docs, doctor, edit, explore, get, help,
    help-search, hook, i, init, install, install-test, it, link,
    list, ln, login, logout, ls, outdated, owner, pack, ping,
    prefix, profile, prune, publish, rb, rebuild, repo, restart,
    root, run, run-script, s, se, search, set, shrinkwrap, star,
    stars, start, stop, t, team, test, token, tst, un,
    uninstall, unpublish, unstar, up, update, v, version, view,
    whoami
npm <command> -h  quick help on <command>
npm -l            display full usage info
npm help <term>   search for help on <term>
npm help npm      involved overview
Specify configs in the ini-formatted file:
    /home/miguel/.npmrc
or on the command line via: npm <command> --key value
Config info can be viewed via: npm help config
npm@6.4.1 /usr/lib/node_modules/npm
miguel@mike-machine:~$ `
Thanks, Miguel de Carvalho

**Comment by MiguelNdeCarvalho** [comment]
it appeared me to put the database password, then exited and got this

miguel@mike-machine:~$ sudo bash /var/lib/dpkg/info/onlyoffice-documentserver.postinst
[sudo] password for miguel: 
Use of uninitialized value $action in string eq at /usr/share/debconf/frontend line 27.
postinst called with unknown argument `'
miguel@mike-machine:~$

**Comment by ShockwaveNN** [comment]
@MiguelNdeCarvalho what's the output of `json --version`

**Comment by ShockwaveNN** [comment]
@MiguelNdeCarvalho ok. seems fine.
try to change first line of `/var/lib/dpkg/info/onlyoffice-documentserver.postinst` to `# !/bin/sh -x` (add -x flag)
and run `sudo bash /var/lib/dpkg/info/onlyoffice-documentserver.postinst configure`

**Comment by MiguelNdeCarvalho** [comment]
I am going to post it in here so I dont spam here. https://del.dog/jokutejago.diff

**Comment by ShockwaveNN** [comment]
@MiguelNdeCarvalho everything seems fine, seems it's the same problem as in https://github.com/ONLYOFFICE/DocumentServer/issues/314, but we didn't find a reason for this. My only advance - try to clean everything up (`/etc/onlyoffice`, `/var/www/onlyoffice`) and reinstall

**Comment by MiguelNdeCarvalho** [comment]
I think somehow it got fixed now. Dont get error when trying to upgrade apt. Thanks for help mate. If I get problem again I will reopen.Really thanks @ShockwaveNN

**Comment by MiguelNdeCarvalho** [comment]
I only has this issue right now xD. `miguel@mike-machine:/var/www/onlyoffice/documentserver/server/SpellChecker$ sudo npm install

npm ERR! err.code.match is not a function

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/miguel/.npm/_logs/2019-01-21T17_21_06_670Z-debug.log
`

**Comment by ShockwaveNN** [comment]
@MiguelNdeCarvalho what's your nodejs version?

**Comment by MiguelNdeCarvalho** [comment]
@ShockwaveNN 
`nodejs -v
v8.15.0`

**Comment by MiguelNdeCarvalho** [comment]
here is the log that It generated https://del.dog/uluporazor.pl

**Comment by ShockwaveNN** [comment]
@MiguelNdeCarvalho this is strange, never seen anything like this error.
Try to remove `node_modules` from `/var/www/onlyoffice/documentserver/server/SpellChecker` and reinstall by `npm install`

**Comment by trofim24** [comment]
@MiguelNdeCarvalho
Try to remove `npm-shrinkwrap.json` and `node_modules` from `/var/www/onlyoffice/documentserver/server/SpellChecker`  and reinstall by `npm install`

**Comment by MiguelNdeCarvalho** [comment]
going to try it guys thanks @trofim24 @ShockwaveNN

**Comment by ShockwaveNN** [comment]
@MiguelNdeCarvalho I google up some other error of `* is not a function` and seems it mostly caused by incapability of versions of nodejs
Are you sure that `node 8` is only installed version of nodejs? 
How did you install it?

**Comment by MiguelNdeCarvalho** [comment]
I installed npm 10.14.0 . Now I am getting this: 
miguel@mike-machine:/var/www/onlyoffice/documentserver/server/SpellChecker$ sudo npm install
Unhandled rejection Error: Command failed: /usr/bin/git clone --depth=1 -q -b master https://git@github.com/ONLYOFFICE/nodehun.git /home/miguel/.npm/_cacache/tmp/git-clone-a91fe1cd
/home/miguel/.npm/_cacache/tmp/git-clone-a91fe1cd/.git: Permission denied

    at ChildProcess.exithandler (child_process.js:294:12)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:962:16)
    at Socket.stream.socket.on (internal/child_process.js:381:11)
    at Socket.emit (events.js:182:13)
    at Pipe._handle.close (net.js:610:12)

npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://npm.community>

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/miguel/.npm/_logs/2019-01-28T10_09_28_608Z-debug.log
miguel@mike-machine:/var/www/onlyoffice/documentserver/server/SpellChecker$

**Comment by ShockwaveNN** [comment]
@MiguelNdeCarvalho I'm not sure it has something with us, seems some permission problems, I don't think you should install any modules to `/home/miguel/.npm`

**Comment by MiguelNdeCarvalho** [comment]
So all is working now @ShockwaveNN . Last thing I am getting is this :( `onlyoffice-documentserver:metrics: ERROR (spawn error)`

**Comment by MiguelNdeCarvalho** [comment]
ah ok. Really thanks @ShockwaveNN for all the help :D

**Comment by morficus** [comment]
In case anyone else runs in to this issue exact same error message, the culprit has to do with npm trying to clone something that is not a valid git reference.

In my case (not using DocumentServer) I was referencing a branch in an external repository that had recently been forced-pushed to. Keep in mind that even tho you put a branch name in your `package.json` file... npm will turn that in to a specific commit hash in `package-lock.json`. Since that specific hash didn't exist... `npm install` (or `npm ci`) is not able to pull data and hence things fail.


TL;DR: check your package.json file to see if it is referencing branches (in external repos) that recently changed or no longer exist.

**Comment by shmakovpn** [comment]
I had the same error 'npm ERR! err.code.match is not a function'. I am using nodejs v10. I copied SpellChecker folder in my user folder. Then I made 'chown -R user:user SpellChecker'. Then I successully ran 'npm install' under my user, moved SpellChecker back into /var/www/onlyoffice/documentserver/server, made 'chown -R onlyoffice:onlyoffice' on it and restart supervisor. Works fine.

**Comment by cyorps** [comment]
Same problem here! Comment from shmakovpn solved the problem.

**Comment by MiguelNdeCarvalho** [comment]
Hey, 
Unfortunately it's me again bothering you @ShockwaveNN . I am running debian on a VM that I have, I installed all following the tutorial and built the dictionary. I change the default port to 500, but it is not working. I used this command `echo onlyoffice-documentserver onlyoffice/ds-port select 500 | sudo debconf-set-selections`
Really thanks,
Miguel de Carvalho

**Comment by ShockwaveNN** [comment]
@MiguelNdeCarvalho Hi, please create separate issue and attach logs from `/var/log/onlyoffice`
